nucypher/blockchain/eth/sol/source/contracts/Issuer.sol

```diff
@@ -11,35 +10,35 @@ import "zeppelin/token/ERC20/SafeERC20.sol";
 
 /**
 * @notice Contract for calculation of issued tokens
-* @dev |v2.1.1|
+* @dev |v3.1.1|
 */
 abstract contract Issuer is Upgradeable {
     using SafeERC20 for NuCypherToken;
-    using SafeMath for uint256;
     using AdditionalMath for uint32;
 
     event Burnt(address indexed sender, uint256 value);
     /// Issuer is initialized with a reserved reward
     event Initialized(uint256 reservedReward);
 
+    uint128 constant MAX_UINT128 = uint128(0) - 1;
+
     NuCypherToken public immutable token;
-    uint256 public immutable totalSupply;
+    uint128 public immutable totalSupply;
 
     uint256 public immutable miningCoefficient;
```

Comment:
```suggestion
    uint256 public immutable mintingCoefficient;
```
We'll replace all the `mining` references with `minting` in the two-phase PR right?

Reply:
right, in next PR